linera-views/src/graphql.rs

```diff
@@ -0,0 +1,853 @@
+impl<C: Context, I: async_graphql::OutputType> async_graphql::OutputType for SetView<C, I>
+where
+    C: Send + Sync,
+    I: serde::ser::Serialize + serde::de::DeserializeOwned + Clone + Send + Sync,
+    ViewError: From<C::Error>,
+{
+    fn type_name() -> Cow<'static, str> {
+        format!("[{}]", I::qualified_type_name()).into()
+    }
+
+    fn qualified_type_name() -> String {
+        format!("[{}]!", I::qualified_type_name())
+    }
+
+    fn create_type_info(registry: &mut async_graphql::registry::Registry) -> String {
+        I::create_type_info(registry);
+        Self::qualified_type_name()
+    }
+
+    async fn resolve(
+        &self,
+        ctx: &async_graphql::ContextSelectionSet<'_>,
+        field: &async_graphql::Positioned<async_graphql::parser::types::Field>,
+    ) -> async_graphql::ServerResult<async_graphql::Value> {
+        let indices = self
+            .indices()
+            .await
+            .map_err(|e| async_graphql::Error::from(e).into_server_error(ctx.item.pos))?;
+        let indices_len = indices.len();
+        async_graphql::resolver_utils::resolve_list(ctx, field, indices, Some(indices_len)).await
+    }
+}
+
+use crate::set_view::CustomSetView;
+#[async_trait::async_trait]
+impl<C: Context, I: async_graphql::OutputType> async_graphql::OutputType for CustomSetView<C, I>
+where
+    C: Send + Sync,
+    I: crate::common::CustomSerialize + Clone + Send + Sync,
+    ViewError: From<C::Error>,
+{
+    fn type_name() -> Cow<'static, str> {
+        format!("[{}]", I::qualified_type_name()).into()
+    }
+
+    fn qualified_type_name() -> String {
+        format!("[{}]!", I::qualified_type_name())
+    }
+
+    fn create_type_info(registry: &mut async_graphql::registry::Registry) -> String {
+        I::create_type_info(registry);
+        Self::qualified_type_name()
+    }
+
+    async fn resolve(
+        &self,
+        ctx: &async_graphql::ContextSelectionSet<'_>,
+        field: &async_graphql::Positioned<async_graphql::parser::types::Field>,
+    ) -> async_graphql::ServerResult<async_graphql::Value> {
+        let indices = self
+            .indices()
+            .await
+            .map_err(|e| async_graphql::Error::from(e).into_server_error(ctx.item.pos))?;
+        let indices_len = indices.len();
+        async_graphql::resolver_utils::resolve_list(ctx, field, indices, Some(indices_len)).await
+    }
+}
+
+use crate::log_view::LogView;
+impl<C: Send + Sync, T: async_graphql::OutputType> async_graphql::TypeName for LogView<C, T> {
+    fn type_name() -> Cow<'static, str> {
+        format!("LogView_{}", T::type_name()).into()
+    }
+}
+#[async_graphql::Object(name_type)]
+impl<C: Context, T: async_graphql::OutputType> LogView<C, T>
+where
+    C: Send + Sync,
+    T: serde::ser::Serialize + serde::de::DeserializeOwned + Clone + Send + Sync,
+    ViewError: From<C::Error>,
+{
+    // TODO this slightly changes the API, but unfortunately function arguments live on
```

Comment:
What's the solution that is waiting to be implemented? Also, comments cannot refer to previous version of the code.

Reply:
I tried to put the parameters on the object proper, but unfortunately that turned out to require significant changes to the `async-graphql` data model, so I decided against it. I should have removed this comment, sorry!